### PR TITLE
Use interop capabilities for JsUuid

### PR DESCRIPTION
**Pull request by lahma**
@viceice here's an example how using TypeReference would work. There's slight change in behavior as value returned from Engine is now `JsUuid` instead of `Guid`. I think it's a bit hard to model primitive types the way behavior was earlier as Jint doesn't know about it and might fail to account for it at some point (even though it was considered a string).

### CODE REVIEW

**Comment by viceice** [comment]
sorry, not yet. had some days off with family

**Comment by lahma** [comment]
Well that's a great reason 👍🏻, no hurry

File: Jint.Tests/Runtime/Domain/JsUuid.cs [hunk 1]
```diff
@@ -3,25 +3,38 @@
...
+
+    public static readonly JsUuid Empty = new JsUuid(Guid.Empty);
+
+    public static JsUuid Parse(JsValue value)
     {
-        internal readonly Guid _value;
-        public static readonly JsUuid Empty = new JsUuid(Guid.Empty);
+        return value switch
+        {
+            { } js when Guid.TryParse(js.AsString(), out var res) => new JsUuid(res),
```
Comment (viceice) [inline Jint.Tests/Runtime/Domain/JsUuid.cs:16]:
```suggestion
            IObjectWrapper ow when ow.Target is JsUuid res => res,
            { } js when Guid.TryParse(js.AsString(), out var res) => new JsUuid(res),
```

can already be a JsUuid. `Uuid.parse(Uuid.Empty)`

File: Jint.Tests/Runtime/UuidTests.cs [hunk 2]
```diff
@@ -28,16 +28,16 @@ private object RunTest(string source)
         [Fact]
         public void Empty()
         {
-            Assert.Equal(Guid.Empty, RunTest($"Uuid.parse('{Guid.Empty}')"));
-            Assert.Equal(Guid.Empty, RunTest($"Uuid.Empty"));
+            Assert.Equal(JsUuid.Empty, RunTest($"Uuid.parse('{Guid.Empty}')"));
+            Assert.Equal(JsUuid.Empty, RunTest($"Uuid.Empty"));
```
Comment (viceice) [inline Jint.Tests/Runtime/UuidTests.cs:32]:
```suggestion
            Assert.Equal(JsUuid.Empty, RunTest($"Uuid.Empty"));
            Assert.Equal(JsUuid.Empty, RunTest($"Uuid.parse(Uuid.Empty)"));
```

File: Jint.Tests/Runtime/UuidTests.cs [hunk 3]
```diff
@@ -9,11 +10,10 @@ public class UuidTests : IDisposable
         private readonly Engine _engine;
 
         public UuidTests()
-        {
-            _engine = new Engine(o => o.AddObjectConverter(new UuidConverter()))
-                .SetValue("copy", new Func<Guid, Guid>(v => new Guid(v.ToByteArray())))
-                ;
-            UuidConstructor.CreateUuidConstructor(_engine);
+        {;
+            _engine = new Engine(options => options.AllowOperatorOverloading());
```
Comment (viceice) [inline Jint.Tests/Runtime/UuidTests.cs:14]: Whats this option for?

File: Jint.Tests/Runtime/UuidTests.cs [hunk 4]
```diff
@@ -9,11 +10,10 @@ public class UuidTests : IDisposable
...
+            _engine.SetValue("copy", new Func<Guid, Guid>(v => new Guid(v.ToByteArray())));
```
Comment (viceice) [inline Jint.Tests/Runtime/UuidTests.cs:15]: is it used?

**Review by viceice — COMMENTED** [review]
Works in general, needed some additional changes for our internal code

**Pull request closed without merging**